[RUM-9617] - Base Benchmark app with support for Test Scenarios and external config (#881)
DataDog/dd-sdk-reactnative: benchmarks/README.md

# Benchmark runner app for Datadog React Native SDK

> # Build and run

Reviewer: ```suggestion
## Build and run
```

- Run yarn install from root folder
- From iOS Folder run pod install

> - Run yarn ios/android from root folder

Reviewer: Making some guesses on how to clarify this, but feel free to take or leave - my main concerns here are ensuring that the sense of sequence is clear here, and ensuring that the commands are clear.
```suggestion
1. From the root folder, run `yarn install`.
2. From the iOS folder, run `pod install`.
3. From the root folder, run `yarn ios` or `yarn android`.
```

> ## Enabling New architcture

Reviewer: ```suggestion
## Enable new architecture
```

### iOS
- From ios folder, run:

> RCT_NEW_ARCH_ENABLED=1 pod install

Reviewer: ````suggestion
- From the iOS folder, run:
```
RCT_NEW_ARCH_ENABLED=1 pod install
```
````

### Android

> Set newArchEnabled=true on benchmarks/android/gradle.properties

Reviewer: ```suggestion
Set `newArchEnabled=true` in `benchmarks/android/gradle.properties`.
```

> # ENV config

Reviewer: ```suggestion
## ENV config
```

> The .env config file holds the Datadog API config, and also allows to set up a test scenario that will be launched directly when the app boots up. If no scenario is provided the app will start without loading any scenario and await for a deeplink to trigger a test scenario.

Reviewer: ```suggestion
The .env config file contains configuration for the Datadog API, as well as a test scenario that you can launch when the app boots up. If it doesn't contain a test scenario, the app waits for a deeplink to trigger a test scenario instead.
```

    DD_CLIENT_TOKEN="CLIENT_TOKEN"
    DD_API_KEY="API_KEY"
    DD_APP_ID="APP_ID"
    DD_SITE="SITE"
    DD_ENV="ENV"
    BENCH_SCENARIO=""
    BENCH_RUN_TYPE=""

> # Changing ENV

Reviewer: ```suggestion
## Change ENV settings
```

### Debug

>      $ ENVFILE=.env.alternate [yarn ios/yarn android]

Reviewer: ````suggestion
```
     // yarn ios/yarn android
     $ ENVFILE=.env.alternate
```
````

### Release 

>     $ ENVFILE=.env.alternate ./gradlew assembleRelease

Reviewer: I'm guessing this is an instruction for everything but iOS...? Can we clarify the use case for this?
````suggestion
```
    $ ENVFILE=.env.alternate ./gradlew assembleRelease
```
````

>     iOS - create schemes and select them during xcodebuild

Reviewer: ```suggestion
In iOS, create schemes and select them during `xcodebuild`.
```

> # Setting test scenarios via Deeplink

Reviewer: ```suggestion
## Set test scenarios using deeplinks
```

> Test scenarios can be triggered by running a deeplink with a specific set of parameters to the benchmark app.

Reviewer: ```suggestion
You can trigger test scenarios by running a deeplink with a specific set of parameters to the benchmark app.
```

> ## Run Test Scenario

Reviewer: ```suggestion
## Run a test scenario
```

> Launches a specific test scenario with a particular configuration.
> 
> - **Method**: start
> - **Parameters**:
>     - **scenario**: name of the test as defined on Scenario enum in types/testConfig.ts
>     - **runType**: `instumented|baseline|profiling`

Reviewer: This feels like it could benefit from an example!
```suggestion
Launch a specific test scenario with a particular configuration:

- **Method**: `start`
- **Parameters**:
    - **scenario**: name of the test as defined on Scenario enum in `types/testConfig.ts`
    - **runType**: `instrumented|baseline|profiling`
```

### iOS
`xcrun simctl openurl booted "benchmark://start?scenario=navigation&runType=instrumented"`

### Android
`adb shell am start -W -a android.intent.action.VIEW -d 'benchmark://start?scenario=navigation\&runType=instrumented' com.benchmarkrunner`

> ## Stop test Scenario

Reviewer: ```suggestion
## Stop a test scenario
```

Stops the scenario currently running.

> - **Method**: stop

Reviewer: ```suggestion
- **Method**: `stop`
```
